subsys/bluetooth/mesh/Kconfig

```diff
@@ -36,17 +36,37 @@ config BT_MESH_PROVISIONER
 	help
 	  Enable this option to have support for provisioning remote devices.
 
-if BT_MESH_PROVISIONER
+config BT_MESH_CDB
+	bool "Mesh Configuration Database"
```

Comment:
You might want to make this "Mesh Configuration Database [EXPERIMENTAL]" (same convention as used for other experimental features) to let us play around with the API in future Zephyr releases without worrying about API stability.